src/cmds/scripts/pbs_habitat.in

```diff
@@ -658,28 +652,38 @@ if [ "${PBS_START_SERVER:-0}" != 0 ] ; then
 	if [ $create_new_svr_data -eq 0 ]; then
 		# datastore directory already exists
 		# do database upgrade activities
-		upgrade_pbs_database
-		ret=$?
-		if [ $ret -ne 0 ]; then
-			if [ $ret -eq 2 ]; then
-				echo "It appears that PostgreSQL has been upgraded independently of PBS Pro."
-				echo "The PBS Pro database must be manually upgraded. Please refer to the"
-				echo "instructions in the release notes for version @PBS_VERSION@ located here:"
-				echo "https://github.com/PBSPro/pbspro/releases"
-			else 
-				echo "Failed to upgrade PBS Datastore"
-			fi
-			exit $ret
-		else
-			if [ -d "${old_inst_dir}" ]; then
-				backupdir "$old_inst_dir" "$PBS_HOME"
-				if [ $? -ne 0 ]; then
-					echo "Failed to backup $old_inst_dir, please follow the below instructions:"
-					echo "*** Backup "$old_inst_dir" if you need to downgrade pgsql later on."
-					echo "*** For future upgrades to be successful run the below command."
-					echo "*** cp -pr ${PBS_EXEC}/pgsql ${PBS_HOME}/pgsql.forupgrade"
+		data_dir="${PBS_HOME}/datastore"
+		if [ ! -f "${data_dir}/PG_VERSION" ]; then
+			echo "Database version file: ${data_dir}/PG_VERSION not found, cannot continue"
+			return 1
+		fi
+		sys_pgsql_ver=$(echo `${PGSQL_BIN}/postgres -V` | awk 'NR==1 {print $NF}' | cut -d '.' -f 1,2)
+		old_pgsql_ver=`cat ${data_dir}/PG_VERSION`
+		if [ "$sys_pgsql_ver" != "$old_pgsql_ver" ]; then
+			# Upgrade postgres
+			upgrade_pbs_database $sys_pgsql_ver $old_pgsql_ver
+			ret=$?
+			if [ $ret -ne 0 ]; then
+				if [ $ret -eq 2 ]; then
+					echo "It appears that PostgreSQL has been upgraded independently of PBS Pro."
+					echo "The PBS Pro database must be manually upgraded. Please refer to the"
+					echo "instructions in the release notes for version @PBS_VERSION@ located here:"
+					echo "https://github.com/PBSPro/pbspro/releases"
```

Comment:
This is about to change to openpbs.  Since that PR is going in next, talk with Hiren for what this should be.

Reply:
@agrawalravi90 As my renaming PR is merged, I think you will get the new text here, but let me know if you need any help.

Reply:
Thanks, I'll take care of it when I resolve conflicts